examples/azure_key/AzureECDsaKey.cs

```diff
@@ -0,0 +1,37 @@
+using Azure.Security.KeyVault.Keys.Cryptography;
+using System;
+using System.Security.Cryptography;
+using AzureSignatureAlgorithm = Azure.Security.KeyVault.Keys.Cryptography.SignatureAlgorithm;
+
+namespace Tmds.Ssh.AzureKeyExample;
+
+sealed class AzureECDsaKey : ECDsa
+{
+    private readonly CryptographyClient _cryptoClient;
+    private readonly ECParameters _publicParameters;
+    private readonly AzureSignatureAlgorithm _signatureAlgorithm;
+
+    public AzureECDsaKey(CryptographyClient client, ECParameters publicParameters, AzureSignatureAlgorithm signatureAlgorithm)
+    {
+        KeySizeValue = publicParameters.Q.X!.Length * 8;
+        _cryptoClient = client;
+        _publicParameters = publicParameters;
+        _signatureAlgorithm = signatureAlgorithm;
+    }
+
+    public override ECParameters ExportParameters(bool includePrivateParameters)
+    {
+        if (includePrivateParameters)
+        {
+            throw new CryptographicException("Cannot export private parameters");
+        }
+
+        return _publicParameters;
+    }
+
+    public override byte[] SignHash(byte[] hash)
+        => _cryptoClient.Sign(_signatureAlgorithm, hash).Signature;
```

Comment:
Can you change this to `_cryptoClient.SignAsync(_signatureAlgorithm, hash, _cancellationToken).GetAwaiter().GetResult().Signature;`, where `_cancellationToken` is the one we get from the `LoadKey` function and gets passed to this class through the constructor?

Reply:
I can, is it a guarantee that the cancellation token is going to be the same going forward?

Reply:
The cancellationtoken is for the scope of the connect, and the key that gets created here has a shorted lifetime than the connect, so it's appropriate to store it in the constructor argument.

Reply:
Sounds good to me